crates/swift-bridge-build/src/package.rs

```diff
@@ -0,0 +1,287 @@
+//! Generate a Swift Package from Rust code
+
+use std::collections::HashMap;
+use std::fs;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::path::Path;
+use std::process::Command;
+
+/// Config for generating Swift packages
+pub struct GeneratePackageConfig<'a> {
+	/// The directory containing the generated bridges
+	pub bridge_dir: &'a dyn AsRef<Path>,
+    /// Path per platform. e.g. `(ApplePlatform::iOS, "target/aarch64-apple-ios/debug/libmy_rust_lib.a")`
+	pub paths: HashMap<ApplePlatform, &'a dyn AsRef<Path>>,
+	/// The directory where the package will be saved
+	pub out_dir: &'a dyn AsRef<Path>,
+	/// The name for the Swift package
+	pub package_name: &'a str
+}
+
+impl<'a> GeneratePackageConfig<'a> {
+	/// Creates a new `GeneratePackageConfig` for generating Swift Packages from Rust code.
+	pub fn new(bridge_dir: &'a dyn AsRef<Path>, paths: HashMap<ApplePlatform, &'a dyn AsRef<Path>>, out_dir: &'a dyn AsRef<Path>, package_name: &'a str) -> Self {
+		Self {
+			bridge_dir,
+			paths,
+			out_dir,
+			package_name
+		}
+	}
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, Eq, PartialEq, Hash)]
+/// Currently supported platforms for genereting Swift Packages.
+pub enum ApplePlatform {
+	/// `aarch64-apple-ios`
+	iOS,
+	/// `x86_64-apple-ios`
+	/// `aarch64-apple-ios-sim`
+	///
+	/// iOS simulator for debugging in XCode's simulator.
+	Simulator,
+	/// `x86_64-apple-darwin`
+	macOS,
+	/// no official Rust target for this platform
+	MacCatalyst,
+	/// `aarch64-apple-tvos`
+	/// `x86_64-apple-tvos`
+	tvOS,
+	/// no official Rust target for this platform
+	watchOS,
+	/// no official Rust target for this platform
+	watchOSSimulator,
+	/// no official Rust target for this platform
+	carPlayOS,
+	/// no official Rust target for this platform
+	carPlayOSSimulator,
+}
+
+impl ApplePlatform {
+	/// The directory name inside of the xcframework for the specified platform.
+	fn dir_name(&self) -> &str {
+		match self {
+			ApplePlatform::iOS => "ios",
+			ApplePlatform::Simulator => "simulator",
+			ApplePlatform::macOS => "macos",
+			ApplePlatform::MacCatalyst => "macos-mac-catalyst",
+			ApplePlatform::tvOS => "tvos",
+			ApplePlatform::watchOS => "watchos",
+			ApplePlatform::watchOSSimulator => "watchos-simulator",
+			ApplePlatform::carPlayOS => "carplay",
+			ApplePlatform::carPlayOSSimulator => "carplay-simulator",
+		}
+	}
+	
+}
+
+/// Generates an xcframework embedded in a Swift Package from the Rust project.
+///
+/// - Also see the [relevant book chapter](https://chinedufn.github.io/swift-bridge/building/swift-packages/index.html)
+/// 
+/// # Parameters
```

Comment:
No need for this. You can move any relevant documentation to the `GeneratePackageConfig` type.

If someone is viewing this they can fire up `rustdoc` and read the documentation for `GeneratePackageConfig` to better understand what it is.